### ISSUE TITLE
HasConversion not used when executing BatchUpdate

### ISSUE THREAD

**Issue by SimonLuckenuik** [borisdj/EFCore.BulkExtensions#195]
Hi,
When executing a BatchUpdate where a property of the entity has an "HasConversion", the conversion is not happening.

Repro code:
``` csharp
using System;
using System.Linq;
using EFCore.BulkExtensions;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Storage.ValueConversion;
using Microsoft.Extensions.Logging;
using Microsoft.Extensions.Logging.Console;

namespace BatchUpdateRepro
{
    class Program
    {
        static void Main(string[] args)
        {
            var dbCtxOptionsBuilder = new DbContextOptionsBuilder<ExampleDbContext>()
                .UseLoggerFactory(new LoggerFactory(new[] {
                    new ConsoleLoggerProvider((_, level) => level >= Microsoft.Extensions.Logging.LogLevel.Debug, true)
                }))
                .EnableSensitiveDataLogging()
                .EnableDetailedErrors()
                .UseSqlServer("Data Source=(localdb)\\MSSQLLocalDB;Initial Catalog=Example;Integrated Security=True", b => b.EnableRetryOnFailure());
            var rand = new Random();
            using (var ctx = new ExampleDbContext(dbCtxOptionsBuilder.Options))
            {
                ctx.Examples.Add(new Example
                {
                    Id = rand.Next(),
                    Value = Guid.NewGuid().ToString("N").ToUpperInvariant()
                });
                ctx.Examples.Add(new Example
                {
                    Id = rand.Next(),
                    Value = Guid.NewGuid().ToString("N").ToUpperInvariant()
                });
                ctx.Examples.Add(new Example
                {
                    Id = rand.Next(),
                    Value = Guid.NewGuid().ToString("N").ToUpperInvariant()
                });
                ctx.SaveChanges();

                var newValue = Guid.NewGuid().ToString("N").ToUpperInvariant();
                var example = ctx.Examples.First();
                example.Value = newValue;
                ctx.SaveChanges();
                //Executed DbCommand(12ms) [Parameters=[@p1='263799922', @p0='bfcb2724-b01c-472a-954a-bfbfcf69cc9f' (Nullable = true)], CommandType='Text', CommandTimeout='30']
                // SET NOCOUNT ON;
                // UPDATE [Example] SET[Value] = @p0
                // WHERE [Id] = @p1;
                // SELECT @@ROWCOUNT;

                ctx.Examples.Where(e => e.Id > 0).BatchUpdate(e => new Example
                {
                    Value = newValue
                });

                // Conversion failed when converting from a character string to uniqueidentifier.

                // Failed executing DbCommand(25ms)[Parameters =[param_0 = 'BFCB2724B01C472A954ABFBFCF69CC9F'(Nullable = false)(Size = 32)], CommandType = 'Text', CommandTimeout = '30']
                // UPDATE [e] SET[e].[Value] = @param_0
                // FROM [Example] AS[e]
                // WHERE [e].[Id] > 0
            }
        }
    }

    public class ExampleDbContext : DbContext
    {
        private readonly ValueConverter<string, Guid> _guidConverter = new ValueConverter<string, Guid>(v => new Guid(v), v => v.ToString("N").ToUpperInvariant());

        public ExampleDbContext(DbContextOptions<ExampleDbContext> options)
            : base(options)
        {
        }

        public DbSet<Example> Examples { get; set; }

        protected override void OnModelCreating(ModelBuilder builder)
        {
            var model = builder.Entity<Example>();
            model.ToTable("Example");
            model.HasKey(e => e.Id);
            model.Property(c => c.Value).HasConversion(_guidConverter);
        }
    }

    public class Example
    {
        public int Id { get; set; }
        public string Value { get; set; }
    }
}

```

Database table:
``` sql
CREATE TABLE [dbo].[Example]
(
	[Id] INT NOT NULL PRIMARY KEY, 
    [Value] UNIQUEIDENTIFIER NULL
)
```

When executing the code, the string is not converted to a uniqueidentifier string and the database is rejecting the request with error: "Conversion failed when converting from a character string to uniqueidentifier."

When doing a regular update with EFCore, I get the following query, where p0 was converted to a valid uniqueidentifier string (bfcb2724-b01c-472a-954a-bfbfcf69cc9f):
``` sql
-- [Parameters=[@p1='263799922', @p0='bfcb2724-b01c-472a-954a-bfbfcf69cc9f' (Nullable = true)], CommandType='Text', CommandTimeout='30']
                SET NOCOUNT ON;
                UPDATE [Example] SET[Value] = @p0
                 WHERE [Id] = @p1;
                SELECT @@ROWCOUNT;
```

Using BatchUpdate, you can see that the string is kept as is in param_0, without the dashes (BFCB2724B01C472A954ABFBFCF69CC9F):
``` sql
                -- Failed executing DbCommand(25ms)[Parameters =[param_0 = 'BFCB2724B01C472A954ABFBFCF69CC9F'(Nullable = false)(Size = 32)], CommandType = 'Text', CommandTimeout = '30']
                 UPDATE [e] SET[e].[Value] = @param_0
                 FROM [Example] AS[e]
                 WHERE [e].[Id] > 0
```

Repro project: [BatchUpdateRepro.zip](https://github.com/borisdj/EFCore.BulkExtensions/files/3272064/BatchUpdateRepro.zip)

Thanks!
Simon

**Comment by borisdj** [comment]
Fixed with NuGet 2.4.9

**Comment by SimonLuckenuik** [comment]
Thank you!
